src/viewport.js

```diff
@@ -835,6 +930,99 @@ export class ViewportBindingVirtual_ {
 
 
 /**
+ * Parses viewport meta value. It usually looks like:
+ * ```
+ * width=device-width,initial-scale=1,minimum-scale=1
+ * ```
+ * @param {string} content
+ * @return {!Object<string, string>}
+ * @private Visible for testing only.
+ */
+export function parseViewportMeta(content) {
+  // Ex: width=device-width,initial-scale=1,minimal-ui
+  let params = Object.create(null);
+  if (!content) {
+    return params;
+  }
+  let pairs = content.split(',');
+  for (let i = 0; i < pairs.length; i++) {
+    let pair = pairs[i];
+    let eqIndex = pair.indexOf('=');
+    let name;
+    let value;
+    if (eqIndex != -1) {
+      name = pair.substring(0, eqIndex).trim();
+      value = pair.substring(eqIndex + 1).trim();
```

Comment:
Yeah, it's nice. I'll redo. Things get a little less "pretty" once you start doing trims, etc.